plugins/azure/virtualmachinescaleset/vmssApprovedExtensions.js

```diff
@@ -0,0 +1,79 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'VM Approved Extensions',
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    description: 'Ensures that approved virtual machine extensions are installed.',
+    more_info: 'Extensions are small applications that provide post-deployment configuration and automation on Azure VMs. Extensions installed should be approved by the organization to meet the organizational security requirements.',
+    recommended_action: 'Uninstall unapproved virtual machine extensions',
+    link: 'https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/overview',
+    apis: ['virtualMachines:listAll', 'virtualMachineExtensions:list'],
```

Comment:
why is this api call same as virtual machines? shouldn't it be related to virtualMachineScaleSets: listAll and likewise for extensions https://learn.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-set-extensions/list


https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmScaleSetName}/extensions?api-version=2023-09-01